morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -705,7 +705,15 @@ public void setQueryFactory(final QueryFactory queryFactory) {
 
         if (wr == null) {
             final Query<T> query = (Query<T>) createQuery(unwrapped.getClass()).filter(Mapper.ID_KEY, id);
-            wr = update(query, new BasicDBObject("$set", dbObj), false, false, wc).getWriteResult();
+
+            BasicDBObject unsetObj = new BasicDBObject();
+            handleUnsetValues(dbObj, unsetObj);
+
+            DBObject updateObj = new BasicDBObject("$set", dbObj);
```

Comment:
Is there a risk that `dbObj` might be empty here? If all changes were moved to `unsetObj` by `handleUnsetValues`... And if so, what would happen? Anything bad?

Reply:
No, tried it and seems to work fine even though `$set` is empty, so I want to keep as is for two reasons:
1. To call `isEmpty()` I would need to cast `dbObj` to a `BasicDBObject` on L686, since `DBObject` doesn't have this method.
2. I want to make sure that `updateObj` is not empty. Having it contain at least a `$set` works even though it's empty. But not sure what happens if the entire `updateObj` is empty. Then there might be problems.